pkg/labels/labels.go

```diff
@@ -613,8 +615,15 @@ func parseSource(str string) (src, next string) {
 // in the form of Source:Key=Value or Source:Key if Value is empty. It also parses short
 // forms, for example: $host will be Label{Key: "host", Source: "reserved", Value: ""}.
 func ParseLabel(str string) *Label {
-	lbl := Label{}
-	src, next := parseSource(str)
+	lbl := parseLabel(str, ':')
+	return &lbl
+}
+
+// parseLabel returns the label representation of the given string by value.
+// For Cilium format 'delim' must be passed in as ':'
+// For k8s format 'delim' must be passed in as '.'
+func parseLabel(str string, delim byte) (lbl Label) {
```

Comment:
Why not returning a Label pointer?

Reply:
Some callers (ultimately the `Has()` and `Get()` do not need the pointer, and returning the label by value reduces the need for heap allocations. This is part of the reason this PR makes those operations faster.